### ISSUE TITLE
Access secrets manager in buildspec results in access denied.

### ISSUE THREAD

**Issue by h5aaimtron** [aws/copilot-cli#2546]
We want to use docker hub to get around the rate limiting issue for image pulls. Currently, when copilot provisions the IAM role, it does not add secrets manager access for the codebuild. Can this be updated to all this behavior as pulling from docker hub and the rate limit issue are pretty common now.

**Comment by Lou1415926** [comment]
This totally makes sense to me. Thank you very much for the suggestion!! 

For the time being, you can modify the build project role so that it has the permission to read your secrets manager resources. You might want to make the modification directly on the resource, as opposed to via modifying CloudFormation template, as the template will get reset next time you run `copilot pipeline update`.

**Comment by h5aaimtron** [comment]
@Lou1415926 Yeah, I modified the IAM role to give it access to the secrets manager, modified the BuildProject adding the environment vars pointing to the secrets, and finally added the docker login in the buildspec. It all worked fine, but yeah I just thought this was cumbersome.

**Comment by efekarakus** [comment]
Related https://github.com/aws/copilot-cli/issues/2755
